### PR TITLE
[FLINK-31373] Fix NPE thrown in ProxyOutput

### PR DESCRIPTION
**Pull request by jiangxin369**
## What is the purpose of the change

Fix NPE thrown in ProxyOutput

## Brief change log

  - Fix NPE thrown in ProxyOutput.

## Does this pull request potentially affect one of the following parts:

  - Dependencies (does it add or upgrade a dependency): (no)
  - The public API, i.e., is any changed class annotated with `@Public(Evolving)`: (no)

## Documentation

  - Does this pull request introduce a new feature? (no)
  - If yes, how is the feature documented? (not applicable)

### CODE REVIEW

**Comment by jiangxin369** [comment]
@zhipeng93 Could you help review this PR?

**Comment by zhipeng93** [comment]
@gaoyunhaii Can you take a look at this PR?

File: flink-ml-iteration/src/main/java/org/apache/flink/iteration/operator/allround/OneInputAllRoundWrapperOperator.java [hunk 1]
```diff
@@ -87,7 +90,9 @@ public void setKeyContextElement(StreamRecord<IterationRecord<IN>> record) throw
     @Override
     public void endInput() throws Exception {
         if (wrappedOperator instanceof BoundedOneInput) {
+            setIterationContextRound(iterationContextRound);
```
Comment (lindong28) [inline flink-ml-iteration/src/main/java/org/apache/flink/iteration/operator/allround/OneInputAllRoundWrapperOperator.java:93]: Would it be simpler to do `setIterationContextRound(Integer.MAX_VALUE)`?

Comment (jiangxin369) [reply flink-ml-iteration/src/main/java/org/apache/flink/iteration/operator/allround/OneInputAllRoundWrapperOperator.java]: Yes, you are right. It's safe and simpler. I've updated and please have another look.

**Comment by lindong28** [comment]
Thanks for the PR. LGTM.

**Pull request merged**